### PR TITLE
Make a ComparableType for comparison of dates.

### PR DESCRIPTION
**Pull request by wgevaert**
Fixes issue 49

### CODE REVIEW

**Comment by wgevaert** [comment]
> This is possible in cypher as well:
> 
> ```
> MATCH (x)
> WHERE [x.y, x.z] = [1, 2]
> RETURN x
> ```
> 
> Yet, I don't believe lists and maps are comparable in the current implementation.

Hmmm, you make a good point, equality should indeed also accept CompositeType besides PropertyType. I'll just add that to this pr

> Also, what is the reasoning for not including `=` and `!=` `IN` and `IS (NOT) null` to comparable type?

I believe these are included; ComparableType extends PropertyType, and these are part of PropertyType (except possibly for is(Not)Null).

**Comment by transistive** [comment]
Perfect! Thank you for the explanation. Tomorrow morning I will be able to finish up the outstanding PR comments, so IS NULL is added as well. 

I am looking forward to seeing this merged!

File: src/Equality.php [hunk 1]
```diff
@@ -34,12 +37,16 @@
 class Equality extends BinaryOperator implements BooleanType
 {
     use BooleanTypeTrait;
+    use ErrorTrait;
 
     /**
      * @inheritDoc
      */
-    public function __construct(PropertyType $left, PropertyType $right, bool $insertParentheses = true)
+    public function __construct(Anytype $left, AnyType $right, bool $insertParentheses = true)
```
Comment (marijnvanwezel) [inline src/Equality.php:45]: Please make sure the doccomment corresponds to the expected types.

**Review by marijnvanwezel — COMMENTED** [review]
Looks good after requested change.

**Review by marijnvanwezel — APPROVED** [review]

**Pull request merged**